filament/backend/src/opengl/PlatformCocoaGL.mm

```diff
@@ -122,14 +129,27 @@
     NSRect currentBounds = [nsView convertRectToBacking:nsView.bounds];
 
     // Check if the view has been swapped out or resized.
+    // updateOpenGLContext() needs to call -clearDrawable if the view was
+    // resized, otherwise we do not want to call -clearDrawable, as in addition
+    // to disassociating the context from the view as the documentation says,
+    // it also does what its name implies and clears the drawable pixels.
+    // This is problematic with multiple windows, but still necessary if the
+    // window has resized.
     if (pImpl->mCurrentView != nsView) {
         pImpl->mCurrentView = nsView;
-        pImpl->updateOpenGLContext(nsView, true);
+        bool needsClear = false;
+        auto rectIter = pImpl->mPreviousSwapChainBounds.find(nsView);
+        if (rectIter == pImpl->mPreviousSwapChainBounds.end() ||
+            !NSEqualSizes(currentBounds.size, rectIter->second.size)) {
+            needsClear = true;
+        }
+        pImpl->updateOpenGLContext(nsView, true, needsClear);
     } else if (!CGRectEqualToRect(currentBounds, pImpl->mPreviousBounds)) {
```

Comment:
Does the `mPreviousBounds` field need to exist?  Seems like this is duplicated data, since the conditional here is equivalent to: `!CGRectEqualToRect(currentBounds, pImpl->mPreviousSwapChainBounds[nsView])`.

Reply:
I believe this was for the case when the context did not need to switch views, but the view had changed size. However, it is no longer needed after the refactor.